lms/djangoapps/certificates/docs/decisions/007-transfer-certificate-signatures-from-mongo.rst

```diff
@@ -0,0 +1,45 @@
+
+Transfer certificate signatures from Mongo
+##########################################
+
+Status
+******
+Proposed
+
+Context
+*******
+As a part of Old Mongo Deprecation work https://github.com/openedx/public-engineering/issues/62
+we are planning to remove the support for c4x assets. After removing support for c4x assets in
+StaticContentServer middleware, the certificate signatories for courses with deprecated IDs won't
+be available because those signatures are stored in course assets.
+
+
+Decision
+********
+To keep access to the certificate signatures will be used external signatures from credentials on WEB/PDF
+certificate rendering.
+
+* credentials API must be extended to provide endpoints for adding and retrieving signatures
+
+  * `CourseCertificateViewSet` API
+
+* added signal to copy signatures from Mongo to Credentials on course publish, certificate or signature update
+* added management command to copy signatures from (OLD or both?) Mongo to Credentials
```

Comment:
I would strongly push for copying signatures out of both Old and Split Mongo. The deprecation of Old Mongo is prompting this work, but Credentials should have its own copy of this data since it's needed for serving certificate information. Also, it'd be weird to special case it for only Old Mongo courses (which are very rare), while having a different code path for Split.

Reply:
@ormsbee Course certs are served out of LMS not Credentials, so are you proposing that we should serve Course Certs out of LMS but globally serve the signatures out of Credentials (for both course and program certs) prior to migrating the Course Certs themselves?

Reply:
@hurtstotouchfire in the ADR we stated that signatures and certificate configurations will be copied to Credentials, but the signatures will be used only when signature from MongoDB is not available (right now this case is specific for Old Mongo courses). The certificate configurations will be pushed to credentials using the management command (planned only for Old Mongo courses), and based on new course certificate update signal (for Split courses).

In the future, we would like to move certificate rendering logic from LMS to Credentials+MFEs, as was proposed in [BD-11](https://github.com/openedx/credentials/issues/1734).
I hope the intermediate step, when we need to get certificates configuration from MongoDB, but signatures from Credentials for Split courses, wouldn't be required.

Reply:
@hurtstotouchfire We are proposing that the complete migration of course certificates is out of scope for this project.  The driver here is to unblock the full deprecation of the Old Mongo Modulestore.  Achieving full deprecation requires moving signature assets and we want to do it in a way aligned with the future vision for all certificates living in the Credentials IDA.